tests/unit/TechRecordsService.unitTest.ts

```diff
@@ -657,6 +655,66 @@ describe("updateTechRecord", () => {
         }
       });
     });
+
+    context("and the user wants to update an archived record", () => {
+      it("should return error 400 Cannot update an archived record", async () => {
+        // @ts-ignore
+        const techRecord: ITechRecordWrapper = cloneDeep(records[31]);
+        const MockDAO = jest.fn().mockImplementation();
+        const mockDAO = new MockDAO();
+        const techRecordsService = new TechRecordsService(mockDAO);
+        const recordToUpdate: any = {
+          techRecord: techRecord.techRecord
+        };
+        try {
+          expect(await techRecordsService.updateTechRecord(recordToUpdate, msUserDetails)).toThrowError();
+        } catch (errorResponse) {
+          expect(errorResponse).toBeInstanceOf(HTTPError);
+          expect(errorResponse.statusCode).toEqual(400);
+          expect(errorResponse.body).toEqual(ERRORS.CannotUpdateArchivedRecord);
+        }
+      });
+    });
+
+    context("and the user wants to update a record which doesn't have the specified statusCode", () => {
+      it("should return error 500 Vehicle has no tech-records with status {statusCode}", async () => {
+        // @ts-ignore
+        const techRecord: ITechRecordWrapper = cloneDeep(records[43]);
+        const MockDAO = jest.fn().mockImplementation();
+        const mockDAO = new MockDAO();
+        const techRecordsService = new TechRecordsService(mockDAO);
+        const recordToUpdate: any = {
```

Comment:
why do you need this? Why can't you just pass techRecord? None of the other stuff  is checked

Reply:
fixed